internal/hcsoci/layers.go

```diff
@@ -17,12 +18,19 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-type vpMemEntry struct {
+type lcowLayerEntry struct {
 	hostPath string
 	uvmPath  string
+	scsi     bool
 }
 
-const scratchPath = "scratch"
+const (
+	scratchPath = "scratch"
+
+	// In LCOW, we can't map to PMEM if a layer VHD is over this size, and
+	// we use SCSI instead.
+	maxPMEMSize = 512 * 1024 * 1024 // 512MB
```

Comment:
This needs to be based on `uvm.vpmemMax` and `uvm.vpmemMaxSizeBytes`. Basically SCSI is used 
```go
if (uvm.currentVpMemCount < uvm.vpmemMax && layer < uvm.vpmemMaxSizeBytes) {
    // use pmem
} else {
    // use scsi
}
```

Reply:
Are these actually populated? I'm not sure they are and default.

Reply:
Well. I *simplified* it. You will need to add `vpmemMaxSizeBytes` if passed use it. If not we know the platform is 512MB. And for the `currentVpMemCount` you need to check the refcount value to determine this against what the user passed as `vpmemMax` on the `CreateOpts` makes sense?

Reply:
Yup, got it now. I'll do a follow-up for this.